### PR TITLE
1.19.2

### PR DESCRIPTION
**Pull request by screret**
thanks for letting me do this!

### CODE REVIEW

**Comment by AlphaMode** [comment]
I'll do some testing in a bit

File: src/main/java/slimeknights/mantle/loot/ReplaceItemLootModifier.java [hunk 1]
```diff
@@ -43,29 +43,10 @@
 import static slimeknights.mantle.loot.AddEntryLootModifier.LOOT_ITEM_FUNCTIONS_CODEC;
 import static slimeknights.mantle.loot.AddEntryLootModifier.LOOT_POOL_ENTRY_CODEC;
 import static slimeknights.mantle.loot.MantleLoot.LOOT_MOD_GSON;
+import static slimeknights.mantle.recipe.helper.ItemOutput.ITEM_OUTPUT_CODEC;
 
 /** Loot modifier to replace an item with another */
 public class ReplaceItemLootModifier extends LootModifier {
-  public static final Codec<ItemOutput> ITEM_OUTPUT_CODEC = Codec.PASSTHROUGH.flatXmap(
```
Comment (AlphaMode) [inline src/main/java/slimeknights/mantle/loot/ReplaceItemLootModifier.java]: Why was this moved and stripped down in ItemOutput?

File: src/main/java/slimeknights/mantle/registration/deferred/FluidDeferredRegister.java [hunk 2]
```diff
@@ -79,7 +80,7 @@ public <F extends SimpleFlowableFluid> FluidObject<F> register(String name, Stri
 
     // create block and bucket, they just need a still supplier
     RegistryObject<LiquidBlock> blockObj = blockRegister.register(name + "_fluid", () -> block.apply(stillDelayed));
-    builder.bucket(itemRegister.register(name + "_bucket", () -> new BucketItem(stillDelayed.get(), ItemProperties.BUCKET_PROPS)));
+    builder.bucket(itemRegister.register(name + "_bucket", () -> new DispensibleBucketItem(stillDelayed.get(), ItemProperties.BUCKET_PROPS)));
```
Comment (AlphaMode) [inline src/main/java/slimeknights/mantle/registration/deferred/FluidDeferredRegister.java:84]: Please refrain from doing things that aren't done upstream. If you want this, make a PR for the 1.19.4 branch please

**Review by AlphaMode — CHANGES_REQUESTED** [review]

**Comment by AlphaMode** [comment]
Why is this based off the 1.18.2 branch?

File: src/main/java/slimeknights/mantle/fluid/attributes/FluidAttributes.java [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+ * Fluid implementations are not required to actively use these properties, nor are objects
+ * interfacing with fluids required to make use of them, but it is encouraged.
+ *
+ * The default values can be used as a reference point for mods adding fluids such as oil or heavy
+ * water.
+ *
+ */
+public class FluidAttributes
+{
+  public static final int BUCKET_VOLUME = 1000;
```
Comment (AlphaMode) [inline src/main/java/slimeknights/mantle/fluid/attributes/FluidAttributes.java:51]: This should point to FluidConstants.BUCKET. It seems I actually forgot to do this in my 1.19.4 port

**Review by AlphaMode — CHANGES_REQUESTED** [review]

File: src/main/java/slimeknights/mantle/registration/deferred/FluidDeferredRegister.java (hunk 2 above)
Comment (screret) [reply src/main/java/slimeknights/mantle/registration/deferred/FluidDeferredRegister.java:84]: I did that because i couldn't get the bucket interaction registration working any other way.

**Comment by screret** [comment]
> Why is this based off the 1.18.2 branch?

this is based on 1.18.2 because I found there were less changes to do there.

**Comment by AlphaMode** [comment]
> > Why is this based off the 1.18.2 branch?
> 
> this is based on 1.18.2 because I found there were less changes to do there.

I'm not really fond of this but I suppose its to late to base this off of 1.19.4

File: src/main/java/slimeknights/mantle/client/model/util/MantleItemLayerModel.java [hunk 4]
```diff
@@ -96,8 +98,8 @@ public BakedModel bake(IModelConfiguration owner, ModelBakery bakery, Function<M
       builder.addAll(getQuadsForSprite(data.color(), data.noTint() ? -1 : i, sprite, transform, data.luminosity(), pixels));
     }
     // transform data
-    ImmutableMap<ItemTransforms.TransformType,Transformation> transformMap = PerspectiveMapWrapper.getTransforms(new CompositeModelState(owner.getCombinedTransform(), modelTransform));
-    return new BakedItemModel(builder.build(), particle, Maps.immutableEnumMap(transformMap), overrides, true, owner.isSideLit());
+    ImmutableMap<ItemTransforms.TransformType,Transformation> transformMap = PerspectiveMapWrapper.getTransforms(owner.getTransforms());
```
Comment (AlphaMode) [inline src/main/java/slimeknights/mantle/client/model/util/MantleItemLayerModel.java:101]: Your code is relying on old 1.18.2 model code that doesn't exist in newer porting lib versions and was left in by mistake I suggest you look at the 1.19.4 version of this class

**Review by AlphaMode — CHANGES_REQUESTED** [review]

**Comment by AlphaMode** [comment]
I assume this was abandoned so I'm going to close this. If you do come back to this please base it off the 1.19.4 and not the 1.18.2 branch which is very broken
